lib/game.rb

```diff
@@ -0,0 +1,69 @@
+class Game
+
+	WIN_COMBINATIONS = [[0,4,8], [6,4,2], [0,1,2], [3,4,5], [6,7,8], [0,3,6], [1,4,7], [2,5,8]]
+
+	attr_accessor :board, :player_1, :player_2
+
+		# Originally tried to make the initialize arguments keywords, but one test would not
+		# pass if this was the case.
+	 def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new )
+	 	@board = board
+			# Computer instance will need access to WIN_COMBINATIONS (via move(board)) to make decisions
+			# on how to move.  This passes WIN_COMBINATIONS to board instance.
+		board.win_combinations = self.class::WIN_COMBINATIONS
+		@player_1 = player_1
+	 	@player_2 = player_2
+	 end
+
+  def turn_count
+		board.cells.count { |square| square != " " }
+  end
+
+  def current_player
+		turn_count.even? ? player_1 : player_2
+  end
+
+	def won?
+		winning_combo = nil
+		WIN_COMBINATIONS.each do |combo|
+			if board.cells[combo[0]] != " " && board.cells[combo[0]] == board.cells[combo[1]] && board.cells[combo[1]] == board.cells[combo[2]]
+				winning_combo = combo
+					#PROBLEM with the test here: I keep getting it right with combo 0, 4, 8.  But this
+					# test here has two combos winning.  It expects 2, 5, 8 to be the winning combo.
+					# Had to move the combos around to make this work.
+			end
+		end
+		winning_combo
+	end
+
+	def full?
+  	board.cells.all? { |cell| cell != " " }
+	end
+
+	def draw?
+		full? && !won?
+	end
+
+  def over?
+  	 won? || draw?
+ 	end
+
+	def winner
+		won? ? board.cells[won?[0]] : nil
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.

Reply:
Feel free to give a shout if anything I say here sounds off.  The comment here took me a while to figure out.  At first, I though this would result in an "unassigned variable" error on account of `won`.  But then I realized the comment is NOT saying `if won == won?`. That would produce the error.  Instead, `if won = won?` does two things:  1) it checks if `won?` returns something truthy.  2) If so, the return value of `won?` is assigned to `won`.  That's pretty slick.